### PR TITLE
Set duplicates flag for hotkeys in HokeyManager

### PR DESCRIPTION
**Pull request by dragunoff**
Follow up to https://github.com/OpenRA/OpenRA/pull/17079#discussion_r324466342

### CODE REVIEW

**Review by abcdefg30 — DISMISSED** [review]
Code changes look good.

File: OpenRA.Game/HotkeyDefinition.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public sealed class HotkeyDefinition
 		public readonly Hotkey Default = Hotkey.Invalid;
 		public readonly string Description = "";
 		public readonly HashSet<string> Types = new HashSet<string>();
-		public bool HasDuplicates = false;
+		public bool HasDuplicates;
```
Comment (pchote) [inline OpenRA.Game/HotkeyDefinition.cs]: Can you please also move the assignment from `SettingsLogic.SaveHotkey` to `HotkeyManager.Set` then make this `{ get; internal set; }`?

Comment (dragunoff) [reply OpenRA.Game/HotkeyDefinition.cs]: Done.

File: OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs [hunk 2]
```diff
@@ -803,7 +800,7 @@ void SaveHotkey()
 			Game.Settings.Save();
 
 			foreach (var hd in modData.Hotkeys.Definitions)
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: This needs to be done inside `Set` for the first hokey. The motivation for all of my requests is that the `SettingsLogic` should not have *anything* to do with updating the duplicate flags!

Comment (dragunoff) [reply OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs]: Understood. I forgot about this one. Will fix it 👍

**Comment by dragunoff** [comment]
Updated with a fix up.

**Comment by pchote** [comment]
Added to the milestone so we can ship the complete hotkey code.

**Review by pchote — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/add63443cef94d55f04784d866599d8fb36ef033...07ce564d35f41301515a0cec274d607a06b89a76)

https://github.com/OpenRA/OpenRA/commit/44af88bb4942a28ec40f25486692d4eb87be0c45

**Pull request merged**